model/task/task.go

```diff
@@ -2033,6 +2033,10 @@ func (t *Task) MergeNewTestResults() error {
 	if t.Archived {
 		id = t.OldTaskId
 	}
+	if !evergreen.IsFinishedTaskStatus(t.Status) || t.Status != evergreen.TaskStarted {
```

Comment:
What happens if the status is success? It seems like this would return nil because t.Status != evergreen.TaskStarted. Is that intentional? Do we only want to return something here if the status is  evergreen.TaskStarted?

Reply:
Good catch, this should be an &&. We want to return here if it's not finished and it's not started.